### PR TITLE
dist/tools/doccheck: Reduce noise

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

This adds central documentation of driver initialization parameters in `driver/foo/include/foo_params.h` where missing for drivers used by the Hamilton board. In addition, the macros to overwrite the default parameters are no longer expected to be documented by adding generic exclude patterns. The reasoning is that having the macros documented once is sufficient.

### Testing procedure

Static tests should pass

### Issues/PRs references

Required to get https://github.com/RIOT-OS/RIOT/pull/19090 in

### CODE REVIEW

File: drivers/hdc1000/include/hdc1000_params.h [hunk 1]
```diff
@@ -28,29 +28,51 @@ extern "C" {
 #endif
 
 /**
- * @name    Set default configuration parameters for the HDC1000 driver
+ * @brief   Default configuration parameters for the HDC1000 driver
```
Comment (maribu) [inline drivers/hdc1000/include/hdc1000_params.h:31]:
```suggestion
 * @name    Default configuration parameters for the HDC1000 driver
```

**Comment by benpicco** [comment]
static tests are not happy yet :confused:

**Comment by maribu** [comment]
Thx for the pointer. It is now green :)

**Review by benpicco — APPROVED** [review]
bors merge

**Comment by benpicco** [comment]
bors merge

**Comment by maribu** [comment]
bors retry

**Comment by maribu** [comment]
Thx :)

**Pull request merged**